ES|QL: Add docs for FORK (#130314)
elastic/elasticsearch: docs/reference/query-languages/esql/_snippets/commands/layout/fork.md

```

The `FORK` processing command creates multiple execution branches to operate
on the same input data and combines the results in a single output table.

**Syntax**

```esql
FORK ( <processing_commands> ) ( <processing_commands> ) ... ( <processing_commands> )
```

**Description**

The `FORK` processing command creates multiple execution branches to operate
on the same input data and combines the results in a single output table. A discriminator column (`_fork`) is added to identify which branch each row came from.

**Branch identification:**
- The `_fork` column identifies each branch with values like `fork1`, `fork2`, `fork3`
- Values correspond to the order branches are defined
- `fork1` always indicates the first branch

**Column handling:**
- `FORK` branches can output different columns
- Columns with the same name must have the same data type across all branches  
- Missing columns are filled with `null` values

**Row ordering:**
- `FORK` preserves row order within each branch
- Rows from different branches may be interleaved
- Use `SORT _fork` to group results by branch

::::{note}
`FORK` branches default to `LIMIT 1000` if no `LIMIT` is provided.
::::

> Limitations and usage:

Reviewer: ```suggestion
**Limitations**
```

## `FORK` [esql-fork]

```yaml {applies_to}
serverless: preview
stack: preview 9.1.0
```

The `FORK` processing command creates multiple execution branches to operate
on the same input data and combines the results in a single output table.

**Syntax**

```esql
FORK ( <processing_commands> ) ( <processing_commands> ) ... ( <processing_commands> )
```

**Description**

The `FORK` processing command feeds the input rows into multiple execution

> branches and outputs the results into a single table, with a discriminator column (`_fork`) to identify which branch each row came from.

Reviewer: keeping same verbiage as opening sentence

> The values of the discriminator column `_fork` are `fork1`, `fork2`, etc. and
> they designate which `FORK` branch the current row comes from.
> The values of the `_fork` column always start with `fork1`, which indicates that
> the row comes from the first branch.
> 
> `FORK` branches can output different columns, but columns with the
> same name must have the same data type across all branches.

Reviewer: ```suggestion
**Branch identification:**
- The `_fork` column contains values `fork1`, `fork2`, etc.
- Values correspond to the order branches are defined
- `fork1` always indicates the first branch
```

When a column does not exist in a `FORK` branch, but it exists in the output of
other branches, `FORK` will add `null` values to the rows that have missing
columns.

`FORK` preserves the order of the rows from each subset, but it does not
guarantee that the rows will follow the same order in which the `FORK` branches
are defined. The rows from the first branch can be interleaved with the rows
from subsequent branches. Use `SORT _fork` after `FORK` if you need to change
this behaviour.

> `FORK` branches receive an implicit `LIMIT 1000` if no `LIMIT` is provided.

Reviewer: ```suggestion
::::{note}
`FORK` branches default to `LIMIT 1000` if no `LIMIT` is provided.
::::
``````
